api/api-rule-violations-known.list

```diff
@@ -148,6 +148,8 @@ API rule violation: names_match,kubevirt.io/api/core/v1,VirtualMachineInstanceSt
 API rule violation: names_match,kubevirt.io/api/core/v1,WatchdogDevice,I6300ESB
 API rule violation: names_match,kubevirt.io/api/instancetype/v1alpha1,VirtualMachineInstancetypeSpec,GPUs
 API rule violation: names_match,kubevirt.io/api/instancetype/v1alpha2,VirtualMachineInstancetypeSpec,GPUs
+API rule violation: names_match,kubevirt.io/api/instancetype/v1beta1,FirmwarePreferences,DeprecatedPreferredUseEfi
+API rule violation: names_match,kubevirt.io/api/instancetype/v1beta1,FirmwarePreferences,DeprecatedPreferredUseSecureBoot
```

Comment:
The field should be marked as deprecated but we cannot change its name. This file should only ever see removals, not additions.

Reply:
The field name hasn't changed, only the internal attribute name in-line with other examples in the codebase and k8s. That's why this name mismatch error is now present.

Reply:
Yeah sorry I mean the variable name

Reply:
Sorry I missed your reply here. I think this is a bad pattern, as implied by the fact that it literally violates the rules...
If we think it makes sense for variables to have a different name than the API field, we should not make that a violation of the rules.
I also don't quite see the point of changing the name of the variable, declaring it deprecated should be enough.
That being said that's just my opinion, I won't push back against this PR since it wouldn't be fair, as others have done the same thing.

Reply:
@jean-edouard no issues I forgot to mention you directly!

I appreciate your points but without this anyone hacking around with this code has to rely on external tooling to pickup on the fact that the field and associated variable is deprecated. Renaming the variable in this way makes it pretty obvious to anyone either reading or extending the code that the field in question is deprecated and likely shouldn't be used going forward.

In the long term I can look at making [`names_match`](https://github.com/kubernetes/kube-openapi/blob/master/pkg/generators/rules/names_match.go) a little more forgiving with `Deprecated` fields so we don't end up with these violations but for now I'd rather keep this as it is.

Reply:
https://github.com/kubevirt/kubevirt/issues/13248 Thanks!